modules/logic_app_set/main.tf

```diff
@@ -40,27 +40,42 @@ resource "azurerm_resource_group_template_deployment" "workflow_deployment" {
   depends_on = [azurerm_logic_app_workflow.workflow]
 }
 
+data "azurerm_monitor_diagnostic_categories" "diagnostic_categories" {
+  for_each    = var.log_analytics_workspace_id == null ? {} : local.logic_app_instances
+  resource_id = azurerm_logic_app_workflow.workflow[each.key].id
+}
+
 // Write logs and metrics to log analytics if specified
 resource "azurerm_monitor_diagnostic_setting" "diagnostic_setting" {
   for_each                       = var.log_analytics_workspace_id == null ? {} : local.logic_app_instances
-  name                           = "${each.key}-logs"
+  name                           = "diag-${each.key}"
   target_resource_id             = azurerm_logic_app_workflow.workflow[each.key].id
   log_analytics_workspace_id     = var.log_analytics_workspace_id
   log_analytics_destination_type = "Dedicated"
 
-  log {
-    category = "WorkflowRuntime"
-    retention_policy {
-      enabled = true
-      days    = var.log_retention_days
+  dynamic "log" {
+    for_each = data.azurerm_monitor_diagnostic_categories.diagnostic_categories[each.key].logs
```

Comment:
Why use each.key as index here while other configurations have fixed index of 0?